src/foundry/common/data/data.mjs/actorData.d.ts

```diff
@@ -0,0 +1,130 @@
+import EmbeddedCollection from '../../abstract/embedded-collection.mjs';
+import {
+  ConfiguredData,
+  ConfiguredDocumentClass,
+  ConfiguredSource,
+  FieldReturnType,
+  PropertiesToSource
+} from '../../../../types/helperTypes';
+import { DocumentData } from '../../abstract/module.mjs';
+import * as documents from '../../documents.mjs';
+import * as fields from '../fields.mjs';
+import { PrototypeTokenData } from '../data.mjs';
+
+interface ActorDataSchema extends DocumentSchema {
+  _id: typeof fields.DOCUMENT_ID;
+  name: typeof fields.REQUIRED_STRING;
+  type: DocumentField<string> & {
+    type: typeof String;
+    required: true;
+    validate: (t: unknown) => boolean;
+    validationError: 'The provided Actor type must be in the array of types defined by the game system';
+  };
+  img: FieldReturnType<typeof fields.IMAGE_FIELD, { default: () => string }>;
+  data: FieldReturnType<typeof fields.OBJECT_FIELD, { default: (data: { type: string }) => any }>; // TODO
+  token: {
+    type: typeof PrototypeTokenData;
+    required: true;
+    default: (data: unknown) => { name: string; img: string };
+  };
+  items: fields.EmbeddedCollectionField<typeof documents.BaseItem>;
+  effects: fields.EmbeddedCollectionField<typeof documents.BaseActiveEffect>;
+  folder: fields.ForeignDocumentField<{ type: typeof documents.BaseFolder }>;
+  sort: typeof fields.INTEGER_SORT_FIELD;
+  permission: typeof fields.DOCUMENT_PERMISSIONS;
+  flags: typeof fields.OBJECT_FIELD;
+}
+
+interface ActorDataBaseProperties {
+  /**
+   * The _id which uniquely identifies this Actor document
+   */
+  _id: string | null;
+
+  /**
+   * The name of this Actor
+   */
+  name: string;
+
+  /**
+   * An Actor subtype which configures the system data model applied
+   */
+  type: string;
+
+  /**
+   * An image file path which provides the artwork for this Actor
+   */
+  img?: string | null;
```

Comment:
This is not optional, as it has a default

Reply:
I'll change that in ItemData too, becuase it's optional there too, despite having a default